src/vpk.cpp

```diff
@@ -12,6 +13,32 @@ namespace VpkParser {
   namespace {
     constexpr uint32_t FILE_SIGNATURE = 0x55aa1234;
     const std::set<uint32_t> SUPPORTED_VERSIONS = {1, 2};
+
+    bool isPathRoot(const std::filesystem::path& path) {
+      return path == "" || path == "/" || path == "\\";
+    }
+
+    std::filesystem::path normalizePath(std::filesystem::path&& path) {
```

Comment:
this is worth a doc comment explaining why we have to perform this ourselves

im also surprised that there's no std filesystem method which can check if a given path is contained within another (or there is in which case we should be using that)